project-base/storefront/pages/order/contact-information.tsx

```diff
@@ -86,6 +87,16 @@ const ContactInformationPage: FC<ServerSidePropsType> = () => {
         }
     }, []);
 
+    useEffect(() => {
+        if (!currentCart.cart?.items.length) {
+            router.replace(cartUrl);
```

Comment:
I would add a return after this one to not overwrite this replace with the following one 🤔

Reply:
This wouldn't happen. But ok, it's improved. I don't like those returns in useEffect since it's always just hacking useEffect which should only return function called when unmount not use return for end of the function run. But I split this logic into two separate useEffects.